bciers/libs/components/src/form/MultiStepBase.test.tsx

```diff
@@ -220,41 +214,28 @@ describe("The MultiStepBase component", () => {
   });
 
   it("submission is disabled if form is still submitting", async () => {
-    useParams.mockReturnValue({
-      formSection: "1",
-      operation: "create",
-    } as QueryParams);
-    const user = userEvent.setup();
-    let resolve: (v: unknown) => void;
-
-    const mockPromiseOnSubmit = vitest.fn().mockReturnValue(
-      new Promise((_resolve) => {
-        resolve = _resolve;
-      }),
-    );
-
     render(
       <MultiStepBase
         {...defaultProps}
         disabled={false}
-        onSubmit={mockPromiseOnSubmit}
+        onSubmit={mockOnSubmit}
+        baseUrl={undefined}
       />,
     );
     const saveAndContinueButton = screen.getByRole("button", {
       name: /Save and Continue/i,
     });
 
-    await user.click(saveAndContinueButton);
-
-    waitFor(() => {
-      expect(saveAndContinueButton).toBeDisabled();
-    });
+    expect(saveAndContinueButton).not.toBeDisabled();
 
-    await act(async () => {
-      resolve(vitest.fn);
-    });
+    fireEvent.click(saveAndContinueButton);
 
-    expect(saveAndContinueButton).not.toBeDisabled();
+    // re-finding the button is necessary for the test to pass
```

Comment:
Oh wow this was the key, I didn't figure this out the other day. Thanks!

Reply:
findByRole is also async while getByRole is sync, so that might be part of it too